libs/designsystem/src/lib/components/dropdown/dropdown-popover.component.integration.spec.ts

```diff
@@ -0,0 +1,163 @@
+  template: '<ng-content></ng-content>',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+class OnPushHostComponent {}
+
+describe('DropdownComponent + PopoverComponent', () => {
+  const items = [
+    { text: 'Item 1', value: 1 },
+    { text: 'Item 2', value: 2 },
+    { text: 'Item 3', value: 3 },
+    { text: 'Item 4', value: 4 },
+    { text: 'Item 5', value: 5 },
+  ];
+  const openDelayInMs = DropdownComponent.OPEN_DELAY_IN_MS;
+
+  describe('by default', () => {
+    const createHost = createHostFactory({
+      component: DropdownComponent,
+      declarations: [
+        ItemComponent,
+        MockComponents(ButtonComponent, IconComponent, IonItem),
+        PopoverComponent,
+        CardComponent,
+      ],
+    });
+
+    let spectator: Spectator<DropdownComponent>;
+    let buttonElement: HTMLButtonElement;
+
+    beforeEach(() => {
+      spectator = createHost(`<kirby-dropdown [usePopover]="true"></kirby-dropdown>`, {
+        props: {
+          items: items,
+        },
+      });
+      buttonElement = spectator.query('button[kirby-button]');
+    });
+    describe('when configured with popout direction', () => {
+      it('open card to the right when popout=right', fakeAsync(() => {
+        spectator.component.popout = HorizontalDirection.right;
+
+        spectator.component.open();
+        tick(openDelayInMs);
+
+        const buttonRect = buttonElement.getBoundingClientRect();
+        const card = spectator.query('kirby-card');
+        const cardRect = card.getBoundingClientRect();
+
+        expect(cardRect.left).toEqual(buttonRect.left);
+      }));
+    });
+
+    describe('when configured with expand=block', () => {
+      beforeEach(() => {
+        spectator.component.expand = 'block';
+        spectator.detectChanges();
+      });
+
+      it('should render button with full width', () => {
+        const componentWidth = spectator.element.clientWidth;
+        const buttonWidth = buttonElement.getBoundingClientRect().width;
+        expect(buttonWidth).toEqual(componentWidth);
+      });
+
+      it('should render dropdown with full width', fakeAsync(() => {
+        // TODO: Move to integration tests
+        spectator.component.ngAfterViewInit();
+
+        spectator.component.open();
+        tick(openDelayInMs);
+
+        const card = spectator.query('kirby-card');
+        const componentWidth = spectator.element.clientWidth;
+        const cardWidth = card.getBoundingClientRect().width;
+        expect(cardWidth).toEqual(componentWidth);
+        expect(card).toHaveComputedStyle({
+          'min-width': '0px',
+          'max-width': 'none',
+        });
+      }));
+    });
+
+    describe('when aligned to right side of viewport', () => {
+      it('should align the dropdown to the right side of button and component container ', (done) => {
+        spectator.element.style.cssFloat = 'right';
+        spectator.component.open();
+        spectator.detectChanges();
+        setTimeout(() => {
```

Comment:
Can we use fakeAsync and tick() instead, like in the other tests, to not have to worry about actual timeouts?